docs/developer/design/authentication.rst

```diff
@@ -0,0 +1,154 @@
+Authentication
+==============
+
+Todo
+----
+
+- Provide uml diagrams for Dex + MetalK8s authentication flow
+
+Open questions
+--------------
+
+- Is support for Dex monitoring and alerting on Grafana
+  something we should think about now?
```

Comment:
Not sure what you mean by that.

- Should we monitor Dex? Yes
- Should we authenticate Prometheus and Grafana through Dex? Not sure, still an open question in the requirements

Reply:
I don't see how that's resolved, the question is still there. If the question is "should we provide a Grafana dashboard to consume monitoring data of Dex?", the answer is still **yes**. That's how we monitor things :)

A question worth asking is **when** will we do this, in your implementation plan. But such questions don't have a place in a design document IMHO, it's more a debate to have either in this peer review or during a design session / team conversation.

So, instead of resolving the conversation and asking the question in the document, just leave it open and start a discussion by mentioning people here, suggesting the various options you thought of and maybe some of the difficulties you anticipate. Once you get a final answer, then you'll know what to write in the document and where, and you'll be able to resolve this conversation. But let's not use merged code for conversation ;)

Reply:
@gdemonet @Ebaneck so to me, having SSO between admin UIs and Grafana is important ...I have added it to the spec. It could be tackled in a dedicated iteration if needs be.